src/app/archive/lib/types.ml

```diff
@@ -0,0 +1,250 @@
+open Core
+open Coda_base
+open Signature_lib
+
+(** Library used to encode and decode types to a graphql format. 
+    
+    Unfortunately, the graphql_ppx does not have an "encode" attribute that allows us to map an OCaml type into some graphql schema type in a clean way. Therefore, we have to make our own encode and decode functions. On top of this, the generated GraphQL schema constructed by Hasura creates insert methods where the input fields are optional, even though the inputs are explicitly labeled as NOT NULL in Postgres. Additionally, graphql_ppx does not let us redefine a new type to make these optional inputs mandatory. Therefore, we are forced to lift the option types to some of our types. Furthermore, some types that are in Postgres but are not GraphQL primitive types are treated as custom scalars. These types include `bigint`, `bit(n)` and enums. graphql_ppx treats custom scalars as Yojson.Basic.t types (usually they are encoded as Json string types). 
```

Comment:
Do you mind line-breaking some of these comments to match the character width of the code? Unless everyone else has word-wrapping turned on an I'm the odd one out.